registry/domain/authentication.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+var path = require('path');
+var strings = require('../../resources/');
+var format = require('stringformat');
+
+var builtin = {
+  basic: {
+    validate: function(authConfig){
+      var isValid = authConfig.username && authConfig.password;
+      return {
+          isValid: isValid,
+          message: isValid ? '' : strings.errors.registry.CONFIGURATION_PUBLISH_BASIC_AUTH_CREDENTIALS_MISSING
+      };
+    },
+    middleware: function(authConfig){
+      var express = require('express');
+      return express.basicAuth(authConfig.username, authConfig.password);
+    }
+  }
+};
+
+var scheme;
+
+module.exports.validate = function(authConfig){
+  if(builtin[authConfig.type]){
+    scheme = builtin[authConfig.type];
+  }
+  else {
+    var cwd = process.cwd();
+    module.paths.push(cwd, path.join(cwd, 'node_modules'));
```

Comment:
Ah! Ok, got it. 

I think I'm more for a hapi-style initialisation in the way we did the plugin, so that you do the require on the initialisation and you solve this problem for free, but I'm ok with this too and we still can revisit later in case we have troubles with npm3.